File: Commune/Commune/Group.swift
//
//  Group.swift
//  Commune
//
//  Created by Mukul Surajiwale on 10/30/16.
//  Copyright © 2016 Mukul Surajiwale. All rights reserved.
//

import UIKit
import Firebase

class Group: NSObject {
	
	let name: String
	let createdBy: FIRUser
	var members: [String] = []
	var groupsRef = FIRDatabase.database().reference(withPath: "Groups")
	
	
	init(name: String, createdBy: FIRUser, members: Array<String>) {
		self.name = name
		self.createdBy = createdBy
		self.members = members
	}
	
	func addGroupToFirebase() -> Bool {
		print(members)
		
		// Add the group to Firebase
		let groupRef = groupsRef.childByAutoId()
		groupRef.setValue(["CreatorUid" : createdBy.uid, "Name" : name])
		groupRef.child("Members").setValue([createdBy.uid : createdBy.displayName])
		
		// Add the group to the user
		let usersRef = FIRDatabase.database().reference(withPath: "Users")
		usersRef.child(createdBy.uid).child("Groups").child(groupRef.key).setValue(["Name" : name])
		
		// Find if each member to add exists in the database. Get the users user id and then add them
		// as a memeber to the group.
		for member in members {
			usersRef.queryOrdered(byChild: "email").queryEqual(toValue: member).observe(.value, with: { snapshot in
				
				if !snapshot.exists() {
					print("USER NOT FOUND")
				}
				
				for child in snapshot.children.allObjects as! [FIRDataSnapshot] {
					print(child.key)
					let dict = child.value as! NSDictionary
					groupRef.child("Members").child(child.key).setValue(dict["name"]!)
					
				}
				
			})
		}
		
		
	
		return true
	}
	
}

File: android/app/src/main/java/xyz/communeapp/commune/Group.java
package xyz.communeapp.commune;

import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

/**
 * Created by Rabi on 11/2/16.
 */

public class Group {
    private String name;
    private FirebaseUser creator;
    private DatabaseReference databaseRef;

    public Group() {

    }

    public Group(String name, FirebaseUser creator, FirebaseDatabase database) {
        this.name = name;
        this.creator = creator;
        this.databaseRef = database.getReference().child("Groups").push();
        databaseRef.child("Name").setValue(name);
        databaseRef.child("CreatorUid").setValue(creator.getUid());
        databaseRef.child("Members").child(creator.getUid()).setValue(creator.getDisplayName());

        database.getReference().child("Users").child(creator.getUid()).child("Groups").push().child("Name").setValue(name);
    }
}
